examples/features/health/client/main.go

```diff
@@ -4,65 +4,98 @@ import (
 	"context"
 	"flag"
 	"fmt"
-	"io"
 	"log"
 	"time"
 
 	"google.golang.org/grpc"
-	healthpb "google.golang.org/grpc/health/grpc_health_v1"
+	pb "google.golang.org/grpc/examples/features/proto/echo"
+	_ "google.golang.org/grpc/health"
+	"google.golang.org/grpc/resolver"
 )
 
-var (
-	addr   = flag.String("addr", "localhost:50051", "the address to connect to")
-	system = "" // empty string represents the health of the system
+var serviceConfig = `{
+	"loadBalancingPolicy": "round_robin",
+	"healthCheckConfig": {
+		"serviceName": ""
+	}
+}`
+
+const (
+	exampleScheme      = "example"
+	exampleServiceName = "health.example.grpc.io"
 )
 
-func callCheck(client healthpb.HealthClient) {
-	response, err := client.Check(context.Background(), &healthpb.HealthCheckRequest{
-		Service: system,
-	})
+var addrs = []string{"localhost:50051", "localhost:50052"}
 
-	if err != nil {
-		log.Fatalf("client.Check(_) = _, %v: ", err)
+func callUnaryEcho(c pb.EchoClient) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	r, err := c.UnaryEcho(ctx, &pb.EchoRequest{})
+	if err == nil {
+		fmt.Println("UnaryEcho: ", r.GetMessage())
 	}
-
-	fmt.Println("Check: ", response.Status)
 }
 
-func callWatch(client healthpb.HealthClient) {
-	watch, err := client.Watch(context.Background(), &healthpb.HealthCheckRequest{
-		Service: system,
-	})
+func main() {
+	flag.Parse()
+
+	address := fmt.Sprintf("%s:///%s", exampleScheme, exampleServiceName)
+
+	options := []grpc.DialOption{
+		grpc.WithInsecure(),
+		grpc.WithBlock(),
+		grpc.WithDefaultServiceConfig(serviceConfig),
+	}
 
+	conn, err := grpc.Dial(address, options...)
 	if err != nil {
-		log.Fatalf("client.Watch(_) = _, %v: ", err)
+		log.Fatalf("did not connect %v", err)
 	}
-	watch.CloseSend()
+	defer conn.Close()
+
+	echoClient := pb.NewEchoClient(conn)
 
 	for {
-		response, err := watch.Recv()
-		if err == io.EOF {
-			break
-		}
-		if err != nil {
-			log.Fatalf("watch.Recv() = _, %v: ", err)
-		}
-
-		fmt.Println("Watch: ", response.Status)
+		callUnaryEcho(echoClient)
 		time.Sleep(time.Second)
 	}
 }
 
-func main() {
-	flag.Parse()
+// Following is an example name resolver implementation. Read the name
```

Comment:
Sorry for not being clearer.  This implementation is not necessary; please use the manual resolver instead to inject the server addresses: https://godoc.org/google.golang.org/grpc/resolver/manual

Let me know if its usage isn't clear and I can help (we also use this in many tests if you would like to see examples).

Reply:
I did notice the manual resolver and tried poking around it for a bit. I will take another look at it and how to use it. I was actually thinking about writing a quick static resolver that would accept a scheme like `static:///addr1,addr2,...`

Reply:
@mjpitz,

To use the manual resolver, I think you can do something like:

```go
r, cleanup := manual.GenerateAndRegisterManualResolver()
defer cleanup()
r.InitialState(resolver.State{Addresses: []resolver.Address{{Addr: ... }, ...})
cc, err := grpc.Dial(r.Scheme()+":///unused")
```

Let me know if you have any problems.  Thanks!

Reply:
Oh, no I figured that. It was just something I thought about writing before.